examples/plot_model_agnostic_importance.py

```diff
@@ -0,0 +1,166 @@
+"""
+Model Agnostic Variable Selection when Linear Models Fail
+=============================================================
+
+In this example, we illustrate the limitations of variable selection methods based on
+linear models using the circles dataset. We first use the distilled conditional
+randomization test (d0CRT), which is based on linear models :footcite:t:`liu2022fast` and then
+demonstrate how model-agnostic methods, such as Leave-One-Covariate-Out (LOCO), can
+identify important variables even when classes are not linearly separable.
+
+To evaluate the importance of a variable, LOCO re-fits a sub-model using a subset of the
+data where the variable of interest is removed. The importance of the variable is
+quantified as the difference in loss between the full model and the sub-model. As shown
+in :footcite:t:`williamson_2021_nonparametric` , this loss difference can be interpreted as an
+unnormalized generalized ANOVA (difference of R²).  Denoting :math:`\mu` the predictive
+model used, :math:`\mu_{-j}` the sub-model where the j-th variable is removed, and
+:math:`X^{-j}` the data with the j-th variable removed, the loss difference can be
+expressed as:
+
+.. math::
+    \psi_{j} = \mathbb{V}(y) \left[ \left[ 1 - \\frac{\mathbb{E}[(y - \mu(X))^2]}{\mathbb{V}(y)} \\right] - \left[ 1 - \\frac{\mathbb{E}[(y - \mu_{-j}(X^{-j}))^2]}{\mathbb{V}(y)} \\right] \\right]
+
+where :math:`\psi_{j}` is the LOCO importance of the j-th variable.
+
+References
```

Comment:
Can you add the Reference section at the end of the example?

Reply:
done